include/seqan3/alignment/configuration/align_config_result.hpp

```diff
@@ -106,11 +157,12 @@ class result : public pipeable_config_element<result<with_type>, with_type>
  * \relates seqan3::align_cfg::result
  * \{
  */
-//!\brief The default constructor defaults to score-only computation.
-result() -> result<detail::with_score_type>;
-
 //!\brief Deduces the alignment result from the given constructor argument.
-template <typename with_type>
-result(with_type) -> result<remove_cvref_t<with_type>>;
+template <typename alignment_result_tag_t>
+result(alignment_result_tag_t) -> result<remove_cvref_t<alignment_result_tag_t>>;
```

Comment:
```suggestion
result(alignment_result_tag_t) -> result<alignment_result_tag_t>;
```
This was wrong before as well and due to missing understanding of const and & deduction. But if you do not bind the parameter at a reference neither const nor & will be used in the type. So it would be less confusing to remove the cvref as it suggests that there might be a ref or const but it is not. 
As far as I see you can then also remove the core_language concept? But I am not sure.